data/src/main/java/com/microsoft/azure/kusto/data/http/HttpClientFactory.java

```diff
@@ -1,108 +1,71 @@
-            final ConnectionKeepAliveStrategy keepAliveStrategy = new CustomConnectionKeepAliveStrategy(properties.maxKeepAliveTime());
-            httpClientBuilder.setKeepAliveStrategy(keepAliveStrategy);
+    public static HttpClient create(HttpClientProperties properties) {
+        LOGGER.info("Creating new HTTP Client");
+        HttpClientOptions options = new HttpClientOptions();
+
+        // If all properties are null, create with default client options
+        if (properties == null) {
+            options.setResponseTimeout(Duration.ofMinutes(10));
+            return HttpClient.createDefault(options);
         }
 
-        if (properties.getPlanner() != null) {
-            httpClientBuilder.setRoutePlanner(properties.getPlanner());
-        }
+        // MS Docs indicate that all setters handle nulls so even if these values are null everything should "just work"
+        // Note that the first discovered HttpClientProvider class is loaded. HttpClientProviders can be swapped in or out
+        // by simply
+        // Docs: https://learn.microsoft.com/en-us/java/api/com.azure.core.util.httpclientoptions?view=azure-java-stable
 
-        if (properties.getProxy() != null) {
-            httpClientBuilder.setProxy(properties.getProxy());
-        }
-
-        if (properties.supportedProtocols() != null) {
-
-            try {
-                httpClientBuilder.setSSLSocketFactory(new SSLConnectionSocketFactory(SSLContext.getDefault(), properties.supportedProtocols(), null,
-                        SSLConnectionSocketFactory.getDefaultHostnameVerifier()));
-            } catch (NoSuchAlgorithmException e) {
-                LOGGER.error("Failed to set supported protocols", e);
-            }
-        }
+        options.setMaximumConnectionPoolSize(properties.maxConnectionTotal());
+        options.setConnectionIdleTimeout(Duration.ofSeconds(properties.maxIdleTime()));
+        options.setResponseTimeout(properties.timeout());
+        options.setHttpClientProvider(properties.provider());
 
-        return httpClientBuilder.build();
-    }
-
-    /**
-     * A custom connection keep-alive strategy that uses the server instructions set in the {@code Keep-Alive}
-     * response header; if the response doesn't contain a {@code Keep-Alive} header, the client will use a configurable
-     * keep-alive period.
-     */
-    static class CustomConnectionKeepAliveStrategy implements ConnectionKeepAliveStrategy {
+        // Set Keep-Alive headers if they were requested.
+        // NOTE: Servers are not obligated to honor client requested Keep-Alive values
+        if (properties.isKeepAlive()) {
+            Header keepAlive = new Header(HttpHeaderName.CONNECTION.getCaseSensitiveName(), "Keep-Alive");
+            // Keep-Alive is Non-standard from the client so core does not have an enum for it
+            Header keepAliveTimeout = new Header("Keep-Alive", "timeout=" + properties.maxKeepAliveTime());
```

Comment:
If the client negotiates HTTP/2, the Keep-Alive header will be stripped because connection specific headers are forbidden in HTTP/2.

HTTP/2 is basically infinite Keep-Alive. A single connection is reused to remove the overhead of establishing new connections